siliconcompiler/schema/schema_cfg.py

```diff
@@ -3557,6 +3440,129 @@ def schema_constraint(cfg, scenario='default', name = 'default'):
+            routing layer.Wildcards ('*') can be used for net names.""")
+
+    scparam(cfg, ['constraint', 'net', name, 'maxlayer'],
+            sctype='str',
+            shorthelp="Constraint: Net maximum routing layer",
+            switch="-constraint_net_maxlayer 'name <str>'",
+            example=[
+                "cli: -constraint_net_maxlayer 'nreset m1'",
+                "api: chip.set('constraint', 'net', 'nreset', 'maxlayer', 'm1')"],
+            schelp="""
+            Maximum metal layer to be used for automated place and route
+            specified on a per net basis. Metal names should either be the PDK
+            specific metal stack name or an integer with '1' being the lowest
+            routing layer.Wildcards ('*') can be used for net names.""")
+
+
+    # AREA
+    scparam(cfg, ['constraint', 'outline'],
+            sctype='[(float,float)]',
+            unit='um',
+            scope='job',
+            shorthelp="Constraint: Layout outline",
+            switch="-constraint_outline <[(float,float)]>",
+            example= ["cli: -constraint_outline '(0,0)'",
+                      "api: chip.set('constraint', 'outline', (0,0))"],
+            schelp="""
+            List of (x,y) points that define the outline physical layout
+            physical design. Simple rectangle areas can be defined with two points,
+            one for the lower left corner and one for the upper right corner. All
+            values are specified in microns.""")
+
+    scparam(cfg, ['constraint', 'corearea'],
+            sctype='[(float,float)]',
+            unit='um',
+            scope='job',
+            shorthelp="Constraint: Layout core area",
+            switch="-constraint_corearea <[(float,float)]>",
+            example= ["cli: -constraint_corearea '(0,0)'",
+                      "api: chip.set('constraint', 'corearea', (0,0))"],
+            schelp="""
+            List of (x,y) points that define the outline of the core area for the
+            physical design. Simple rectangle areas can be defined with two points,
+            one for the lower left corner and one for the upper right corner. All
+            values are specified in microns.""")
+
+    scparam(cfg, ['constraint', 'coremargin'],
+            sctype='float',
+            unit='um',
+            scope='job',
+            shorthelp="Constraint: Layout core margin",
+            switch="-constraint_coremargin <float>",
+            example= ["cli: -constraint_coremargin 1",
+                      "api: chip.set('constraint', 'coremargin', '1')"],
+            schelp="""
+            Halo/margin between the outline and core area for fully
+            automated layout sizing and floorplanning.
+            Corearea outline length = outline length * coremargin,
+            where coremaging is a number between 1 (no halo) and 0
+            (no core area). This number is ignored if corearea is
+            supplied.""")
```

Comment:
Correct this to be coremargin in um again. If you want to keep the number is ignored, we can set the defvalue to 0.0

Reply:
fixed